biggerquery/beam_manager.py

```diff
@@ -0,0 +1,205 @@
+import avro.schema
+from apache_beam.io import BigQueryDisposition
+
+import avro.schema
+import apache_beam as beam
+from apache_beam.io.gcp.internal.clients import bigquery
+from apache_beam.options.pipeline_options import \
+    PipelineOptions, SetupOptions, StandardOptions, GoogleCloudOptions, WorkerOptions
+
+
+class BeamManager:
+    """
+    Used for read/write operations. It is base class for further operations.
+    """
+    def __init__(self):
+        self.pipeline = None
+
+    def read_from_big_query(self, dataset, project_id, standard_sql, query):
+        return beam.io.Read(beam.io.BigQuerySource(
+            dataset=dataset,
+            project=project_id,
+            use_standard_sql=standard_sql,
+            query=query
+        ))
+
+    def read_from_avro(self, file_source_path):
+        return beam.io.ReadFromAvro(file_source_path)
+
+    def write_to_avro(self, file_output_path, schema):
+        return beam.io.WriteToAvro(
+            file_output_path,
+            avro.schema.parse(schema),
+            num_shards=1,
+            shard_name_template='S_N',
+            file_name_suffix='.avro')
+
+
+class TemplatedBeamManager:
+    """
+    Decorator that resolves table names/all kind of variables.
+    """
+
+    def __init__(self, beam_manager, internal_tables, external_tables, run_datetime, extras):
+        self.beam_manager = beam_manager
+        self.extras = extras
+        self.internal_tables = internal_tables
+        self.external_tables = external_tables
+        self.run_datetime = run_datetime
+
+    def read_from_big_query(self, dataset, project_id, standard_sql, query):
+        query = query.format(**self._template_variables())
+        return self.beam_manager.read_from_big_query(dataset, project_id, standard_sql, query)
+
+    def read_from_avro(self, file_source_path):
+        return self.beam_manager.read_from_avro(file_source_path)
+
+    def write_to_avro(self, file_output_path, schema):
+        return self.beam_manager.write_to_avro(file_output_path, schema)
+
+    def write_truncate_to_big_query(self, table_name, schema):
+        table_name = self._legacy_table_name(self._template_variables()[table_name]) + '$' + self.run_datetime.replace('-', '')
+        return beam.io.WriteToBigQuery(
+            table_name,
+            schema=schema,
+            write_disposition=BigQueryDisposition.WRITE_TRUNCATE)
+
+    def _legacy_table_name(self, table_name):
+        spl = table_name.split('.')
+        return spl[0] + ':' + '.'.join(spl[1:len(spl)])
+
+    @property
+    def run_datetime(self):
```

Comment:
run_datetime -> run_datetime_str, I know its bad, but let's be consistent with the bigquery dataset manager

Reply:
Look now please.